### ISSUE TITLE
image slideshow in UITableViewCell

### ISSUE THREAD

**Issue by DominikButz** [zvonicek/ImageSlideshow#131]
Hi, 
strictly speaking this is not an issue but a question. Maybe anyone stumbled over the same problem. I have added the image slideshow as subview to a UITableViewCell (dynamic). The problem is that somehow it cancels the user's tap on the table view cell, so the segue to the detail view is not triggered....   
Important: I have not added a tap gesture recogniser to trigger full screen mode. 
thanks,
Dominik

**Comment by htarwara6245** [comment]
That is easy one, and I have made it. Let me tell you how did I solve it. I just used Delegate Pattern.

At your cell, you have to implement like that.

`import UIKit
import ImageSlideshow

protocol NewDetailMainPhotoCellDelegate {
    func slideShowTapped(cell: NewDetailMainPhotoCell)
}

class NewDetailMainPhotoCell : UITableViewCell{
    
    @IBOutlet var slideshow: ImageSlideshow!
    
    var delegate: NewDetailMainPhotoCellDelegate?
    
    func cellTapped(sender: AnyObject) {
        delegate?.slideShowTapped(cell: self)
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        slideshow.backgroundColor = UIColor.white
        slideshow.slideshowInterval = 3.0
        slideshow.pageControlPosition = PageControlPosition.insideScrollView
        slideshow.pageControl.currentPageIndicatorTintColor = UIColor.white
        slideshow.pageControl.pageIndicatorTintColor = UIColor(red: 43.0/255.0, green: 163.0/255.0, blue: 67.0/255.0, alpha: 1.0)
        slideshow.contentScaleMode = UIViewContentMode.scaleAspectFill
        
        self.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(NewDetailMainPhotoCell.cellTapped)))
        // Initialization code
    }
    
    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)
        
        // Configure the view for the selected state
    }
}`

And then at your UIViewController tableview delegate method, cellForRowAtIndexPath

`func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "NewDetailMainPhotoCell") as! NewDetailMainPhotoCell
        cell.slideshow.setImageInputs(detailImages)
        cell.delegate = self
        return cell
    }`

Then add the following function at that view controller,

`func slideShowTapped(cell: NewDetailMainPhotoCell) {
        cell.slideshow.presentFullScreenController(from: self)
    }`

And make sure your view controller which include uitableview extends `NewDetailMainPhotoCellDelegate`


Good Luck, Happy Coding.

**Comment by htarwara6245** [comment]
Please show your issue more specifically or if that worked out, please close the issue

**Comment by DominikButz** [comment]
Thank you very much!  Issue resolved - I have implemented the suggested delegate pattern and it's working now. 
In each table view cell, the  image slideshow occupies the whole content view. See the attached screenshot - the problem was when I tapped on the image slide show the segue to the detail view didn't get triggered. It did get triggered however when tapping on the labels that are on top of the image slideshow!

 This is not about the full screen view controller: I didn't add a tap gesture recogniser to the image slideshow in the table view cell. I could have done that and with the same delegate pattern just trigger the same function delegateSlideshow Tapped instead of adding a gesture recogniser to the whole cell as you have proposed but that works without any issue, too. 

The table view has single selection set so I guess this is not the issue. 

Thanks

![screen shot 2017-03-19 at 15 15 07](https://cloud.githubusercontent.com/assets/5699871/24078959/193984b8-0cb7-11e7-88a8-08cb675f3833.png)
